lib/ThirdPage.dart

```diff
@@ -0,0 +1,228 @@
+import 'package:flutter/material.dart';
+import 'package:flutterproject/SecondPage.dart';
+import 'package:flutterproject/databasehelper.dart';
+
+class ThirdPage extends StatefulWidget {
+  const ThirdPage({Key? key});
+
+  @override
+  _ThirdPageState createState() => _ThirdPageState();
+}
+
+class _ThirdPageState extends State<ThirdPage> {
+  final dbHelper = DatabaseHelper();
+  List<Map<String, dynamic>> dataList = [];
+
+  @override
+  void initState() {
+    super.initState();
+    _getData();
+  }
+
+  Future<void> _getData() async {
+    dataList = await dbHelper.getAllData();
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Colors.lightBlue,
+        title: const Center(
+          child: Text("MİRSAD ASSİSTANT", style: TextStyle(fontWeight: FontWeight.bold)),
+        ),
+        leading: IconButton(
+          icon: const Icon(Icons.arrow_circle_left_outlined, color: Colors.white),
+          iconSize: 30,
+          onPressed: () {
+            Navigator.pop(context);
+          },
+        ),
+      ),
+      body: DecoratedBox(
+        decoration: const BoxDecoration(
+          image: DecorationImage(
+            image: AssetImage('photos/code.jpg'),
+            fit: BoxFit.cover,
+          ),
+        ),
+        child: Padding(
+          padding: const EdgeInsets.all(20.0),
+          child: Scrollbar(
+            child: GridView.builder(
+              gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                crossAxisCount: 2,
+                mainAxisSpacing: 10,
+                crossAxisSpacing: 10,
+                childAspectRatio: 1,
+              ),
+              itemCount: dataList.length,
+              itemBuilder: (context, index) {
+                return GestureDetector(
+                  onTap: () {
+                    _navigateToSecondPage(dataList[index]);
```

Comment:
navigation işlemlerini buradaki gibi ayrı bir metotta yapıp onu çağırabilirsin

Reply:
Yapıldı .Sadece birinci sayfafnın decoratedBox'undaki  ve Thirdpage sayfasındaki SecondPage'ye geçiş için eklenilen metodta hata verdi. 2 tane manuel geçiş kalmış oldu.

Reply:
Düzeltildi.